### PR TITLE
chore: upgraded to Next.js 14

### LINKED ISSUE #2021: Feature: Upgrade to Next.js 14
### Type of feature

📦 Chore

### Current behavior

We're on Next.js 13.4.7 (older version of Next.js 13)

### Suggested solution

Upgrade to Next.js 14 for speed and stability improvements. We're still using the pages router so maybe stability is less relevant (more along the lines of server actions).

But, we can also try `next dev -- turbo`.

More on all this at https://nextjs.org/blog/next-14

Ensure that Storybook gets the useSWC flag turned on in [main.js](https://github.com/open-sauced/app/pull/2244/files#diff-cafe2123a72c4ce3a9f7e9ee4b0e188256eb02d7ec1e54fa2cadfac99b22f92bR15-R19).

```
  framework: {
    name: "@storybook/nextjs",
    options: { builder: { useSWC: true } },
  },
```

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by nickytonline**
## Description

The deploy preview appears to be all good. I've navigated around the app a bit and things seem to be behaving correctly. Also trying out `next dev --turbo` for faster local development.

Take the deploy preview for a spin. If folks are happy with the smoke test, we can merge and monitor it in the beta environment for a bit before going to prod.

## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [x] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

Closes #2021

## Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -123,8 +123,8 @@
     "@storybook/addon-interactions": "^7.2.0",
     "@storybook/addon-links": "^7.2.0",
     "@storybook/addon-styling": "^1.3.5",
-    "@storybook/nextjs": "^7.2.0",
-    "@storybook/react": "^7.2.0",
+    "@storybook/nextjs": "^7.5.2",
+    "@storybook/react": "^7.5.2",
```
Comment (nickytonline) [inline package.json:127]: This was required as Storybook builds were failing.

**Comment by bdougie** [comment]
Only checking from phone, but get an internal service error when visiting the following link.

https://deploy-preview-2020--oss-insights.netlify.app/hub/lists/new

**Comment by OgDev-01** [comment]
Also getting an error viewing a list 
image

**Comment by OgDev-01** [comment]
This behaviour is seen on all pages that uses `getServersideProps`. I'll look deeper to see why

**Comment by nickytonline** [comment]
> This behaviour is seen on all pages that uses `getServersideProps`. I'll look deeper to see why

Don't spend too much time on that as this isn't a high priority atm. Not sure why I didn't see that. 🤔

**Comment by nickytonline** [comment]
Going to close this for now and revisit this later.

**Comment by nickytonline** [comment]
```
2:55:43 PM: ❯ Using Next.js Runtime - v5.0.0-rc.4
2:55:44 PM: ​
2:55:44 PM: @netlify/plugin-nextjs (onPreBuild event)                     
2:55:44 PM: ────────────────────────────────────────────────────────────────
2:55:44 PM: ​
2:55:44 PM: No Next.js cache to restore
2:55:44 PM: ​
2:55:44 PM: (@netlify/plugin-nextjs onPreBuild completed in 8ms)
2:55:44 PM: ​
2:55:44 PM: Build command from Netlify app                                
2:55:44 PM: ────────────────────────────────────────────────────────────────
2:55:44 PM: ​
2:55:44 PM: $ npm run prebuild && npm run build
2:55:44 PM: npm info using npm@10.2.4
2:55:44 PM: npm info using node@v18.19.1
2:55:44 PM: > @open-sauced/app@2.7.1-beta.2 prebuild
2:55:44 PM: > node generate-netlify-env.js
2:55:44 PM: Running in Netlify CI environment
2:55:44 PM: Running in Netlify deploy-preview environment
2:55:44 PM: npm info ok
2:55:44 PM: npm info using npm@10.2.4
2:55:44 PM: npm info using node@v18.19.1
2:55:44 PM: > @open-sauced/app@2.7.1-beta.2 prebuild
2:55:44 PM: > node generate-netlify-env.js
2:55:45 PM: Running in Netlify CI environment
2:55:45 PM: Running in Netlify deploy-preview environment
2:55:45 PM: > @open-sauced/app@2.7.1-beta.2 build
2:55:45 PM: > next build
2:55:45 PM: ⚠ No build cache found. Please configure build caching for faster rebuilds. Read more: https://nextjs.org/docs/messages/no-cache
2:55:45 PM:    ▲ Next.js 14.1.3
```

**Comment by nickytonline** [comment]
I have to dig a bit, but with v5 RC 4, the deploy went well and the site for the most part appears to be working, but our middleware does not appear to be working. It looks like it may not have been deployed.

Also, locally I'm getting some errors, so maybe I need to update some other things.

CleanShot 2024-03-13 at 15 03 27

**Comment by MarcL** [comment]
@nickytonline - I've brought it up with the team and we'll have a look too.

**Comment by ascorbic** [comment]
Hey @nickytonline! I've just tried this [on a fork](https://github.com/ascorbic/open-sauced-app/pull/1) and [it works](https://deploy-preview-1--glowing-alpaca-a86556.netlify.app/) fine as far as I can see. I added a log inside the middleware and I can see it being called.

**Comment by nickytonline** [comment]
I came back to the site after it just redeployed and now I can hit a workspaces page, so I think this might be some app code that caused my issue. This loads fine now, https://deploy-preview-2020--oss-insights.netlify.app/workspaces/3432aaf7-55db-4c2b-9fde-f61604471e21 .

CleanShot 2024-03-14 at 09 00 40

I think this might be an issue in our code. 😅 cc: @brandonroberts I think there might be  some client-side redirect wonkiness after logging in from the root as I got the generic greedy route initially (see above).

I think for the time being we're all good. I'll continue to test things out @ascorbic @MarcL.

**Comment by nickytonline** [comment]
@brandonroberts, it looks like our `/workspaces` isn't redirecting to the last workspace accessed anymore and ends up redirecting to `/workspaces/dashboard/filter/recent` instead. I'll have to look into that. It doesn't appear to be happening on beta, so maybe I messed something up during a rebase.

TLDR; it's probably me. 😅

**Comment by ascorbic** [comment]
That's great! Let us know how you get on

**Comment by nickytonline** [comment]
The proof of concept with this demonstrated that everything is working with the release candidate of the Netlify Next Runtime. I will update this PR once the official version is released and updated automatically for our sites on Netlify @brandonroberts.

**Comment by nickytonline** [comment]
We're upgrading now. I believe I need to upgrade Storybook as well. Just taking a peek at that.

**Comment by nickytonline** [comment]
Just looking into a Node.js out of memmory error during the build, /workspaces/60c3ee25-2bc6-41fd-babf-d4e294332298. Locally it builds fine, but my machine has more RAM. Going to clear the build cache and restart a new build.

**Comment by nickytonline** [comment]
Looks like I'm getting Node.js out of memory consistently during a build.

```
3:52:40 PM: <--- Last few GCs --->
3:52:40 PM: [5857:0x5b89850]    73690 ms: Mark-sweep 2001.1 (2083.9) -> 2001.1 (2079.6) MB, 438.4 / 0.0 ms  (average mu = 0.429, current mu = 0.128) allocation failure; scavenge might not succeed
3:52:40 PM: [5857:0x5b89850]    74490 ms: Mark-sweep 2010.8 (2089.4) -> 2010.8 (2105.4) MB, 785.1 / 0.0 ms  (average mu = 0.202, current mu = 0.018) allocation failure; scavenge might not succeed
3:52:40 PM: <--- JS stacktrace --->
3:52:40 PM: FATAL ERROR: Reached heap limit Allocation failed - JavaScript heap out of memory
3:52:40 PM:  1: 0xb9a330 node::Abort() [node]
3:52:40 PM:  2: 0xaa07ee  [node]
3:52:40 PM:  3: 0xd71ed0 v8::Utils::ReportOOMFailure(v8::internal::Isolate*, char const*, bool) [node]
3:52:40 PM:  4: 0xd72277 v8::internal::V8::FatalProcessOutOfMemory(v8::internal::Isolate*, char const*, bool) [node]
3:52:40 PM:  5: 0xf4f635  [node]
3:52:40 PM:  6: 0xf61b0d v8::internal::Heap::CollectGarbage(v8::internal::AllocationSpace, v8::internal::GarbageCollectionReason, v8::GCCallbackFlags) [node]
3:52:40 PM:  7: 0xf3c1fe v8::internal::HeapAllocator::AllocateRawWithLightRetrySlowPath(int, v8::internal::AllocationType, v8::internal::AllocationOrigin, v8::internal::AllocationAlignment) [node]
3:52:40 PM:  8: 0xf3d5c7 v8::internal::HeapAllocator::AllocateRawWithRetryOrFailSlowPath(int, v8::internal::AllocationType, v8::internal::AllocationOrigin, v8::internal::AllocationAlignment) [node]
3:52:40 PM:  9: 0xf1db40 v8::internal::Factory::AllocateRaw(int, v8::internal::AllocationType, v8::internal::AllocationAlignment) [node]
3:52:40 PM: 10: 0xf155b4 v8::internal::FactoryBase<v8::internal::Factory>::AllocateRawWithImmortalMap(int, v8::internal::AllocationType, v8::internal::Map, v8::internal::AllocationAlignment) [node]
3:52:40 PM: 11: 0xf17948 v8::internal::FactoryBase<v8::internal::Factory>::NewRawTwoByteString(int, v8::internal::AllocationType) [node]
3:52:40 PM: 12: 0x130438d v8::internal::Runtime_StringBuilderConcat(int, unsigned long*, v8::internal::Isolate*) [node]
3:52:40 PM: 13: 0x1710739  [node]
3:52:42 PM: Aborted (core dumped)
```

Going to try an earlier version of Next.js 14 to see if the result is the same. Locally no issues running `npm run build`.

**Comment by nickytonline** [comment]
A local build via `ntl build` runs fine, but I also have more RAM.

```
❯ ntl build

Netlify Build                                                 
────────────────────────────────────────────────────────────────

❯ Version
  @netlify/build 29.36.1

❯ Flags
  dry: false
  offline: false

❯ Current directory
  /Users/nicktaylor/dev/work/app

❯ Config file
  /Users/nicktaylor/dev/work/app/netlify.toml

❯ Context
  production

❯ Installing plugins
   - @netlify/plugin-nextjs@5.2.2

❯ Using Next.js Runtime - v5.2.2

@netlify/plugin-nextjs (onPreBuild event)                     
────────────────────────────────────────────────────────────────


(@netlify/plugin-nextjs onPreBuild completed in 4ms)

Build command from Netlify app                                
────────────────────────────────────────────────────────────────

$ npm run prebuild && npm run build
npm info using npm@10.2.4
npm info using node@v20.11.0

> @open-sauced/app@2.26.0-beta.2 prebuild
> node generate-netlify-env.js

npm info ok 
npm info using npm@10.2.4
npm info using node@v20.11.0

> @open-sauced/app@2.26.0-beta.2 prebuild
> node generate-netlify-env.js


> @open-sauced/app@2.26.0-beta.2 build
> next build

  ▲ Next.js 14.2.3
  - Environments: .env


./pages/_app.tsx
96:6  Warning: React Hook useEffect has a missing dependency: 'router.events'. Either include it or remove the dependency array.  react-hooks/exhaustive-deps

./pages/feed/index.tsx
115:6  Warning: React Hook useEffect has a missing dependency: 'signIn'. Either include it or remove the dependency array.  react-hooks/exhaustive-deps
123:6  Warning: React Hook useEffect has a missing dependency: 'followersRepo'. Either include it or remove the dependency array.  react-hooks/exhaustive-deps
129:6  Warning: React Hook useEffect has a missing dependency: 'openSingleHighlight'. Either include it or remove the dependency array.  react-hooks/exhaustive-deps

./pages/start.tsx
73:6  Warning: React Hook useEffect has a missing dependency: 'router'. Either include it or remove the dependency array.  react-hooks/exhaustive-deps

./pages/user/notifications.tsx
63:6  Warning: React Hook useEffect has missing dependencies: 'fetchNotifications' and 'router'. Either include them or remove the dependency array.  react-hooks/exhaustive-deps

./pages/workspaces/[workspaceId]/contributor-insights/[listId]/highlights.tsx
163:6  Warning: React Hook useEffect has a missing dependency: 'contributor'. Either include it or remove the dependency array.  react-hooks/exhaustive-deps

./pages/workspaces/[workspaceId]/contributor-insights/new.tsx
97:6  Warning: React Hook useEffect has missing dependencies: 'contributorIds' and 'title'. Either include them or remove the dependency array.  react-hooks/exhaustive-deps

./pages/workspaces/[workspaceId]/repository-insights/new.tsx
53:6  Warning: React Hook useEffect has a missing dependency: 'repos'. Either include it or remove the dependency array.  react-hooks/exhaustive-deps

./pages/workspaces/new.tsx
64:6  Warning: React Hook useEffect has missing dependencies: 'descriptionQuery', 'nameQuery', and 'reposQuery'. Either include them or remove the dependency array.  react-hooks/exhaustive-deps

./components/AddContributorsHeader/add-contributors-header.tsx
39:6  Warning: React Hook useEffect has a missing dependency: 'onSearch'. Either include it or remove the dependency array. If 'onSearch' changes too often, find the parent component that defines it and wrap that definition in useCallback.  react-hooks/exhaustive-deps

./components/Repositories/AddToWorkspaceDrawer.tsx
62:11  Warning: Using `<img>` could result in slower LCP and higher bandwidth. Consider using `<Image />` from `next/image` to automatically optimize images. This may incur additional usage or cost from your provider. See: https://nextjs.org/docs/messages/no-img-element  @next/next/no-img-element

./components/Repositories/AddToWorkspaceModal.tsx
63:17  Warning: Using `<img>` could result in slower LCP and higher bandwidth. Consider using `<Image />` from `next/image` to automatically optimize images. This may incur additional usage or cost from your provider. See: https://nextjs.org/docs/messages/no-img-element  @next/next/no-img-element

./components/Workspaces/InsightUpgradeModal.tsx
29:6  Warning: React Hook useEffect has missing dependencies: 'posthog' and 'workspaceId'. Either include them or remove the dependency array.  react-hooks/exhaustive-deps

./components/Workspaces/TrackedContributorsWizard/SelectFollowingStep.tsx
29:6  Warning: React Hook useEffect has a missing dependency: 'onToggleContributor'. Either include it or remove the dependency array. If 'onToggleContributor' changes too often, find the parent component that defines it and wrap that definition in useCallback.  react-hooks/exhaustive-deps

./components/Workspaces/TrackedRepositoryFilter.tsx
15:6  Warning: React Hook useEffect has a missing dependency: 'handleSelect'. Either include it or remove the dependency array. If 'handleSelect' changes too often, find the parent component that defines it and wrap that definition in useCallback.  react-hooks/exhaustive-deps

./components/Workspaces/WorkspaceBanner.tsx
14:6  Warning: React Hook useEffect has missing dependencies: 'posthog' and 'workspaceId'. Either include them or remove the dependency array.  react-hooks/exhaustive-deps

./components/Workspaces/WorkspaceWelcomeModal.tsx
19:11  Warning: Using `<img>` could result in slower LCP and higher bandwidth. Consider using `<Image />` from `next/image` to automatically optimize images. This may incur additional usage or cost from your provider. See: https://nextjs.org/docs/messages/no-img-element  @next/next/no-img-element

./components/atoms/SkeletonLoader/skeleton-wrapper.stories.tsx
16:14  Warning: The story should use PascalCase notation: tripleWrappers  storybook/prefer-pascal-case

./components/molecules/AuthSection/auth-section.tsx
50:6  Warning: React Hook useEffect has a missing dependency: 'router'. Either include it or remove the dependency array.  react-hooks/exhaustive-deps
56:6  Warning: React Hook useEffect has a missing dependency: 'userInfo'. Either include it or remove the dependency array.  react-hooks/exhaustive-deps

./components/molecules/ContributorHighlight/contributor-highlight-card.tsx
166:6  Warning: React Hook useEffect has a missing dependency: 'taggedRepoList'. Either include it or remove the dependency array.  react-hooks/exhaustive-deps
475:6  Warning: React Hook useEffect has a missing dependency: 'updateSuggestionsDebounced'. Either include it or remove the dependency array.  react-hooks/exhaustive-deps

./components/molecules/ContributorProfileHeader/contributor-profile-header.tsx
396:6  Warning: React Hook useEffect has missing dependencies: 'handleAddToList' and 'selectedList.length'. Either include them or remove the dependency array.  react-hooks/exhaustive-deps

./components/molecules/HighlightInput/highlight-input-form.tsx
317:6  Warning: React Hook useEffect has a missing dependency: 'fetchAllUserHighlights'. Either include it or remove the dependency array.  react-hooks/exhaustive-deps
353:6  Warning: React Hook useEffect has missing dependencies: 'handleGenerateHighlightSummary' and 'taggedRepoList'. Either include them or remove the dependency array.  react-hooks/exhaustive-deps
550:6  Warning: React Hook useEffect has a missing dependency: 'updateSuggestionsDebounced'. Either include it or remove the dependency array.  react-hooks/exhaustive-deps

./components/molecules/HubContributorsHeader/hub-contributors-header.tsx
50:6  Warning: React Hook useEffect has missing dependencies: 'contributorSearch' and 'onSearch'. Either include them or remove the dependency array. If 'onSearch' changes too often, find the parent component that defines it and wrap that definition in useCallback.  react-hooks/exhaustive-deps

./components/molecules/TableHeader/table-header.tsx
58:6  Warning: React Hook React.useEffect has a missing dependency: 'updateSuggestionsDebounced'. Either include it or remove the dependency array.  react-hooks/exhaustive-deps

./components/organisms/GitHubTeamSyncDialog/github-team-sync-dialog.tsx
61:6  Warning: React Hook useEffect has a missing dependency: 'loadTeams'. Either include it or remove the dependency array.  react-hooks/exhaustive-deps

./components/organisms/InsightPage/InsightPage.tsx
131:6  Warning: React Hook useEffect has missing dependencies: 'user?.user_metadata.sub', 'workspaceId', and 'workspacesLoading'. Either include them or remove the dependency array.  react-hooks/exhaustive-deps
528:6  Warning: React Hook useEffect has a missing dependency: 'updateSuggestionsDebounced'. Either include it or remove the dependency array.  react-hooks/exhaustive-deps

./components/organisms/SearchDialog/search-dialog.tsx
47:6  Warning: React Hook useEffect has a missing dependency: 'setOpenSearch'. Either include it or remove the dependency array.  react-hooks/exhaustive-deps
81:6  Warning: React Hook useEffect has missing dependencies: 'cursor', 'providerToken', and 'searchTerm.length'. Either include them or remove the dependency array.  react-hooks/exhaustive-deps
198:6  Warning: React Hook useEffect has a missing dependency: 'setOpenSearch'. Either include it or remove the dependency array.  react-hooks/exhaustive-deps

./components/organisms/TopNav/top-nav.tsx
89:13  Warning: Using `<img>` could result in slower LCP and higher bandwidth. Consider using `<Image />` from `next/image` to automatically optimize images. This may incur additional usage or cost from your provider. See: https://nextjs.org/docs/messages/no-img-element  @next/next/no-img-element

./components/organisms/UserRepositoryRecommendations/user-repository-recommendations.tsx
74:6  Warning: React Hook useEffect has a missing dependency: 'getRepoFullNameByInterests'. Either include it or remove the dependency array.  react-hooks/exhaustive-deps
78:6  Warning: React Hook useEffect has a missing dependency: 'refreshUserRecommendedRepos'. Either include it or remove the dependency array.  react-hooks/exhaustive-deps

./components/shared/AppSidebar/AppSidebar.tsx
220:23  Warning: Using `<img>` could result in slower LCP and higher bandwidth. Consider using `<Image />` from `next/image` to automatically optimize images. This may incur additional usage or cost from your provider. See: https://nextjs.org/docs/messages/no-img-element  @next/next/no-img-element

./lib/hooks/useFilterPrefetch.ts
24:6  Warning: React Hook useEffect has a missing dependency: 'filterOptions'. Either include it or remove the dependency array.  react-hooks/exhaustive-deps

./lib/hooks/useSession.ts
48:6  Warning: React Hook useEffect has missing dependencies: 'getSession', 'loadSession', and 'setStoreData'. Either include them or remove the dependency array.  react-hooks/exhaustive-deps

./lib/hooks/useSupabaseAuth.ts
40:6  Warning: React Hook useEffect has missing dependencies: 'loadSession' and 'store'. Either include them or remove the dependency array.  react-hooks/exhaustive-deps

info  - Need to disable some ESLint rules? Learn more here: https://nextjs.org/docs/basic-features/eslint#disabling-rules
 ✓ Linting and checking validity of types    
   Creating an optimized production build ...
error - No Sentry auth token configured. Source maps will not be uploaded.
You can find information on how to generate a Sentry auth token here: https://docs.sentry.io/api/auth/
After generating a Sentry auth token, set it via the SENTRY_AUTH_TOKEN environment variable during the build.

 ✓ Compiled successfully
 ✓ Collecting page data    
   Generating static pages (0/8)  [    ]You are on the server
You are on the server
You are on the server
You are on the server
You are on the server
You are on the server
You are on the server
You are on the server
 ✓ Generating static pages (8/8)
 ✓ Collecting build traces    
 ✓ Finalizing page optimization    

Route (pages)                                                               Size     First Load JS
┌ ƒ /                                                                       315 B           257 kB
├   /_app                                                                   0 B             257 kB
├ ○ /404                                                                    4.21 kB         303 kB
├ ○ /500                                                                    1.89 kB         261 kB
├ ○ /account-deleted                                                        502 B           288 kB
├ ƒ /api/delete-account                                                     0 B             257 kB
├ ƒ /api/fetchOGData                                                        0 B             257 kB
├ ƒ /explore/topic/[pageId]/[toolName]                                      610 B           949 kB
├ ○ /explore/topic/[pageId]/[toolName]/filter/[...selectedFilter]           606 B           949 kB
├ ○ /feed                                                                   277 B           510 kB
├ ƒ /feed/[id]                                                              555 B           511 kB
├ ƒ /s/[org]/[repo]                                                         98.3 kB         446 kB
├ ƒ /signin                                                                 334 B           257 kB
├ ƒ /signout                                                                335 B           257 kB
├ ƒ /star-search                                                            20.1 kB         357 kB
├ ƒ /star-search/waitlist                                                   4.67 kB         339 kB
├ ○ /start                                                                  8.14 kB         309 kB
├ ƒ /user/[username]                                                        38.1 kB         883 kB
├ ƒ /user/[username]/card                                                   3.4 kB          303 kB
├ ○ /user/notifications                                                     6.59 kB         303 kB
├ ○ /user/settings                                                          10.3 kB         351 kB
├ ƒ /workspaces/[workspaceId]                                               21.7 kB         546 kB
├ ƒ /workspaces/[workspaceId]/activity                                      29.2 kB         394 kB
├ ƒ /workspaces/[workspaceId]/contributor-insights                          9.96 kB         361 kB
├ ƒ /workspaces/[workspaceId]/contributor-insights/[listId]                 385 B           257 kB
├ ƒ /workspaces/[workspaceId]/contributor-insights/[listId]/activity        14.6 kB         528 kB
├ ƒ /workspaces/[workspaceId]/contributor-insights/[listId]/edit            9.36 kB         368 kB
├ ƒ /workspaces/[workspaceId]/contributor-insights/[listId]/highlights      16.2 kB         490 kB
├ ƒ /workspaces/[workspaceId]/contributor-insights/[listId]/overview        15.4 kB         886 kB
├ ƒ /workspaces/[workspaceId]/contributor-insights/new                      8.96 kB         368 kB
├ ƒ /workspaces/[workspaceId]/contributors                                  9.37 kB         882 kB
├ ƒ /workspaces/[workspaceId]/repository-insights                           5.52 kB         347 kB
├ ƒ /workspaces/[workspaceId]/repository-insights/[insightId]/activity      12.3 kB         534 kB
├ ƒ /workspaces/[workspaceId]/repository-insights/[insightId]/contributors  5.96 kB         895 kB
├ ƒ /workspaces/[workspaceId]/repository-insights/[insightId]/dashboard     21.7 kB         536 kB
├ ƒ /workspaces/[workspaceId]/repository-insights/[insightId]/edit          9.15 kB         368 kB
├ ƒ /workspaces/[workspaceId]/repository-insights/new                       9 kB            368 kB
├ ƒ /workspaces/[workspaceId]/settings                                      19.8 kB         395 kB
└ ƒ /workspaces/new                                                         9.32 kB         368 kB
+ First Load JS shared by all                                               277 kB
  ├ chunks/framework-56eb74ff06128874.js                                    45.2 kB
  ├ chunks/main-f4a17ecc0c9d5004.js                                         36.9 kB
  ├ chunks/pages/_app-e59ecb79bc89499b.js                                   172 kB
  ├ css/bdca3dd7443aa63f.css                                                20.1 kB
  └ other shared chunks (total)                                             2.37 kB

ƒ Middleware                                                                80.4 kB

○  (Static)   prerendered as static content
ƒ  (Dynamic)  server-rendered on demand

npm info ok 

(build.command completed in 37.9s)

@netlify/plugin-nextjs (onBuild event)                        
────────────────────────────────────────────────────────────────


(@netlify/plugin-nextjs onBuild completed in 2.5s)

Functions bundling                                            
────────────────────────────────────────────────────────────────

The Netlify Functions setting targets a non-existing directory: netlify/functions

Packaging Functions from .netlify/functions-internal directory:
 - ___netlify-server-handler/___netlify-server-handler.mjs


(Functions bundling completed in 2.6s)

Edge Functions bundling                                       
────────────────────────────────────────────────────────────────

Packaging Edge Functions from .netlify/edge-functions directory:
 - ___netlify-edge-handler-middleware

Packaging Edge Functions from netlify/edge-functions directory:
 - card
 - repo-page-card
 - workspaces-card

(Edge Functions bundling completed in 3s)

@netlify/plugin-nextjs (onPostBuild event)                    
────────────────────────────────────────────────────────────────


Warning: some headers have syntax errors:

Path should come before header "https"

Path should come before header "X-Robots-Tag"

Path should come before header "http"

Path should come before header "X-Robots-Tag"

Path should come before header "https"

Path should come before header "X-Robots-Tag"

Path should come before header "http"

Path should come before header "X-Robots-Tag"

Path should come before header "https"

Path should come before header "X-Robots-Tag"

Path should come before header "http"

Path should come before header "X-Robots-Tag"

(@netlify/plugin-nextjs onPostBuild completed in 727ms)

@netlify/plugin-nextjs (onSuccess event)                      
────────────────────────────────────────────────────────────────


(@netlify/plugin-nextjs onSuccess completed in 684ms)

@netlify/plugin-nextjs (onEnd event)                          
────────────────────────────────────────────────────────────────


(@netlify/plugin-nextjs onEnd completed in 596ms)

Save updated config                                           
────────────────────────────────────────────────────────────────


(options.onEnd completed in 5ms)

Netlify Build Complete                                        
────────────────────────────────────────────────────────────────

(Netlify Build completed in 51.1s)
```

**Comment by nickytonline** [comment]
I'm currently investigating file sizes in the `.netlify` folder to see if anything stands out. I can't upload the result of `find . -type f -exec stat -f '%N %.1f KB' '{}' +` though. It makes the comment too big. 😅 

Same for the `.next` folder when I run `find . -type f -exec stat -f '%N %.1f KB' '{}' +`.

**Comment by nickytonline** [comment]
So it looks like 14.1.1 works fine, but 14.2.2 and 14.2.3 both cause out of memory error issues @MarcL @ascorbic. Again, locally via `ntl build` but no dice on Netlify.com for a deploy. Just FYI.

**Comment by nickytonline** [comment]
> So it looks like 14.1.1 works fine, but 14.2.x versions cause out of memory error issues @MarcL @ascorbic. Again, locally via `ntl build` all good, but no dice on Netlify.com for a deploy. Just FYI.
> 
> I could allocate more memory to Node.js, but that seems kind of hacky since it only starts happening in 14.2.x.

Looks like I could go as high as 14.1.4 for now.

**Comment by MarcL** [comment]
@nickytonline - We'll have a look at the runtime to see what's up.

**Comment by pieh** [comment]
> So it looks like 14.1.1 works fine, but 14.2.x versions cause out of memory error issues @MarcL @ascorbic. Again, locally via `ntl build` all good, but no dice on Netlify.com for a deploy. Just FYI.
> 
> I could allocate more memory to Node.js, but that seems kind of hacky since it only starts happening in 14.2.x.

The OOM I looked at ( https://app.netlify.com/sites/oss-insights/deploys/663d2b98d286670008400d65#L2275-L2293 ) seems to happen during `next build` (in particular during `Creating an optimized production build` step which should be creating various bundles) and at this point runtime didn't do much except for restoring cache and setting some env vars, so at first glance it looks more like somehow Next.js need more resources with new version (?)

This is totally random but seeing some Sentry logs before OOM maybe this could be related https://github.com/getsentry/sentry-javascript/issues/10468 ? Worth a try using their `disableServerWebpackPlugin ` workaround with latest `next` to at least "try something"?

PS. Hi Nick! :)

**Review by bdougie — APPROVED** [review]
Let's move this forward and move the 14.2 debugging to a separate issues.

**Comment by nickytonline** [comment]
> Let's move this forward and move the 14.2 debugging to a separate issue.

Yeah agreed. We don't need to be on the latest 14.x version anyway.

**Review by jpmcb — APPROVED** [review]

**Comment by nickytonline** [comment]
> So it looks like 14.1.1 works fine, but 14.2.x versions cause out of memory error issues @MarcL @ascorbic. Again, locally via `ntl build` all good, but no dice on Netlify.com for a deploy. Just FYI.
> 
> I could allocate more memory to Node.js, but that seems kind of hacky since it only starts happening in 14.2.x.

It looks like 14.2.4 is fine for deploys now. I ran a deploy preview and the build didn't run out of memory like it was previously.

To update, run `npm install next@latest react@latest react-dom@latest`


We should be good but check it what happens in https://github.com/netlify/next-runtime/issues/2471 affects us since we're on the pages router and use middleware.
